src/main/webapp/app/admin/cleanup-service/cleanup-service.component.ts

```diff
@@ -0,0 +1,108 @@
+import { Component, OnInit } from '@angular/core';
+import dayjs from 'dayjs/esm';
+import { CleanupOperation } from 'app/admin/cleanup-service/cleanup-operation.model';
+import { CleanupServiceExecutionRecordDTO, DataCleanupService } from 'app/admin/cleanup-service/cleanup-service.service';
+import { convertDateFromClient, convertDateFromServer } from 'app/utils/date.utils';
+import { Subject } from 'rxjs';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+
+@Component({
+    selector: 'jhi-cleanup-service',
+    templateUrl: './cleanup-service.component.html',
+})
+export class CleanupServiceComponent implements OnInit {
+    private dialogErrorSource = new Subject<string>();
+    dialogError = this.dialogErrorSource.asObservable();
+
+    constructor(private cleanupService: DataCleanupService) {}
+
+    cleanupOperations: CleanupOperation[] = [
+        {
+            name: 'deleteOrphans',
+            deleteFrom: dayjs().subtract(12, 'months'),
+            deleteTo: dayjs().subtract(6, 'months'),
+            lastExecuted: undefined,
+        },
+        {
+            name: 'deletePlagiarismComparisons',
+            deleteFrom: dayjs().subtract(12, 'months'),
+            deleteTo: dayjs().subtract(6, 'months'),
+            lastExecuted: undefined,
+        },
+        {
+            name: 'deleteNonRatedResults',
+            deleteFrom: dayjs().subtract(12, 'months'),
+            deleteTo: dayjs().subtract(6, 'months'),
+            lastExecuted: undefined,
+        },
+        {
+            name: 'deleteOldRatedResults',
+            deleteFrom: dayjs().subtract(12, 'months'),
+            deleteTo: dayjs().subtract(6, 'months'),
+            lastExecuted: undefined,
+        }
+    ];
+
+    ngOnInit(): void {
+        this.loadLastExecutions();
+    }
+
+    loadLastExecutions(): void {
+        this.cleanupService.getLastExecutions().subscribe((executionRecordsBody: HttpResponse<CleanupServiceExecutionRecordDTO[]>) => {
+            const executionRecords = executionRecordsBody.body!;
+            if (executionRecords && executionRecords.length > 0) {
+                this.cleanupOperations.forEach((operation, index) => {
+                    const executionRecord = executionRecords[index];
+                    if (executionRecord && executionRecord.executionDate) {
+                        operation.lastExecuted = convertDateFromServer(executionRecord.executionDate);
+                    }
+                });
+            }
+        });
+    }
+
+    executeCleanupOperation(operation: CleanupOperation): void {
+        const deleteFrom = convertDateFromClient(operation.deleteFrom)!;
+        const deleteTo = convertDateFromClient(operation.deleteTo)!;
```

Comment:
As I see it, this conversion method is typically used in the service. Can you perform the date conversion in the Cleanup Service instead? I feel like that would be more intuitive than passing a string to each of the service methods